src/libraries/libraries-packages.proj

```diff
@@ -14,15 +10,15 @@
   <ItemGroup>
     <ProjectReference Include="$(PkgDir)*\*.proj" Exclude="$(PkgDir)test\*" />
     <ProjectReference Include="$(MSBuildThisFileDirectory)*\src\*.*proj"
-                      Condition="('$(BuildAllConfigurations)' == 'true' or '$(DotNetBuildFromSource)' == 'true')"
+                      Condition="'$(BuildAllConfigurations)' == 'true'"
                       Exclude="$(MSBuildThisFileDirectory)*\src\**\*.shproj"/>
   </ItemGroup>
 
   <!-- Need the PackageIndexFile file property from baseline.props -->
   <Import Project="$(PkgDir)baseline\baseline.props" />
 
   <!-- Calculate PackageDownload items for restore. -->
-  <Import Project="$(RepositoryEngineeringDir)restore\harvestPackages.targets" Condition="'$(MSBuildRestoreSessionId)' != ''" />
+  <Import Project="$(RepositoryEngineeringDir)restore\harvestPackages.targets" Condition="'$(MSBuildRestoreSessionId)' != '' and '$(DotNetBuildFromSource)' != 'true'" />
```

Comment:
@ViktorHofer - I see you removed the 0027-Remove-libraries-packages.proj-harvesting.patch with https://github.com/dotnet/runtime/pull/53294/commits/6c2e477f7848b2fcc12896d25d15855c6ff6391b#diff-a195cee0207dfe976258d38145c8edb053bbe098930d1b701b36aaba78aa9855.  This breaks source-build with:

```
  /repos/runtime/artifacts/source-build/self/src/eng/restore/harvestPackages.targets(12,5): error MSB4062: The "GetLastStablePackage" task could not be loaded from the assembly /repos/runtime/artifacts/source-build/self/package-cache/microsoft.dotnet.build.tasks.packaging/6.0.0-beta.21311.3/tools/net5.0/Microsoft.DotNet.Build.Tasks.Packaging.dll. Could not load file or assembly '/repos/runtime/artifacts/source-build/self/package-cache/microsoft.dotnet.build.tasks.packaging/6.0.0-beta.21311.3/tools/net5.0/Microsoft.DotNet.Build.Tasks.Packaging.dll'. The system cannot find the path specified. [/repos/runtime/artifacts/source-build/self/src/Build.proj]
  /repos/runtime/artifacts/source-build/self/src/eng/restore/harvestPackages.targets(12,5): error MSB4062:  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [/repos/runtime/artifacts/source-build/self/src/Build.proj]
```

I added the changes in the patch to fix source-build.  What were you expecting to be the patch resolution?

Reply:
I just fixed this in main an hour ago: https://github.com/dotnet/runtime/blob/6474c725025a8b29b4265c4d9485018982ee6b16/src/libraries/libraries-packages.proj#L25.